Sources/SnapshotTesting/SnapshotTesting.swift

```diff
@@ -116,3 +108,57 @@ private var trackSnapshots = {
     print("Found \(count) stale snapshot\(count == 1 ? "" : "s"):\n\n\(list)")
   }
 }()
+
+public protocol SnapshotStringConvertible {
+  var snapshotDescription: String { get }
+}
+
+extension NSObject: SnapshotStringConvertible {
+  public var snapshotDescription: String {
+    return self.debugDescription
+      .replacingOccurrences(of: ": 0x[\\da-f]+", with: "", options: .regularExpression)
+  }
+}
+
+private func snap<T>(_ value: T, name: String? = nil, indent: Int = 0) -> String {
+  let indentation = String(repeating: " ", count: indent)
+  let mirror = Mirror(reflecting: value)
+  let count = mirror.children.count
+  let bullet = count == 0 ? "-" : "▿"
+
+  let description: String
+  switch (value, mirror.displayStyle) {
+  case (_, .collection?):
+    description = count == 1 ? "1 element" : "\(count) elements"
+  case (_, .dictionary?):
+    description = count == 1 ? "1 key/value pair" : "\(count) key/value pairs"
+  case (_, .set?):
+    description = count == 1 ? "1 member" : "\(count) members"
+  case (_, .tuple?):
+    description = count == 1 ? "(1 element)" : "(\(count) elements)"
+  case (_, .optional?):
+    let subjectType = String(describing: mirror.subjectType)
+      .replacingOccurrences(of: " #\\d+", with: "", options: .regularExpression)
```

Comment:
In Swift 4.1, nested types sometimes become, _e.g._ `User #1` instead of `User`. This removes that. I thought maybe you could do something clever like use backticks to make a type name weird, but that's not allowed, so I think this is p bulletproof.